src/Grinder/Commands.fs

```diff
@@ -298,7 +302,10 @@ module Processing =
                 let! requestsResult = 
                     [for chat in botSettings.ChatsToMonitor.Set do
                         for user in usernames do
-                            yield ApiExt.restrictUser context.UpdateContext chat user time]
+                            if userCanBeBanned botSettings user then
+                                yield ApiExt.restrictUser context.UpdateContext chat user time
+                            else
+                                yield async.Return (Result.Error <| sprintf "Cannot ban admin @%s" user) ]
```

Comment:
This will send message `botSettings.ChatsToMonitor.Set` times instead of once. We can invert chat -> user loop to user -> chat and check if user is admin before iterating over chats.

Also user should be checked [below](https://github.com/Liminiens/grinder/blob/master/src/Grinder/Commands.fs#L449)